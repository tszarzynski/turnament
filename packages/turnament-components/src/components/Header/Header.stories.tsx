import Header from "./Header";

export default {
	component: Header,
	title: "Header",
};

export const Default = () => {
	return <Header>Title</Header>;
};
