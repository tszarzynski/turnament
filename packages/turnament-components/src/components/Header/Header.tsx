import Hr from "../Hr";

type Props = {
	children: React.ReactNode;
	turnamentType?: string;
	playersNum?: number;
	minPtsToWin: number;
	roundsPlayed: number;
	minRounds?: number;
	matchesPlayed: number;
	minMatches: number;
	minGames: number;
	maxGames: number;
	gamesPlayed: number;
};

function valueOrPlaceholder(value?: string | number) {
	return value || "";
}

const Header = ({
	children,
	turnamentType,
	playersNum,
	minPtsToWin,
	minRounds,
	minMatches,
	roundsPlayed,
	matchesPlayed,
	minGames,
	maxGames,
	gamesPlayed,
}: Props) => {
	const headings = [
		"Type",
		"Players",
		"Pts To Win",
		"Rounds",
		"Matches",
		"Games",
	];

	const columns = [
		valueOrPlaceholder(turnamentType),
		valueOrPlaceholder(playersNum),
		valueOrPlaceholder(minPtsToWin),
		minRounds ? `${roundsPlayed} / ${minRounds}` : "",
		minMatches ? `${matchesPlayed} / ${minMatches}` : "",
		minGames ? `${gamesPlayed} / ${minGames}-${maxGames}` : "",
	];

	return (
		<div className="static border-secondary border-t-2 p-0.5 ">
			<div className="py-6 text-center font-bold text-4xl text-secondary uppercase leading-none">
				{children}
			</div>

			<Hr />
			<table className="w-full table-fixed border-separate border-spacing-0.5">
				<thead>
					<tr>
						{headings.slice(0, 3).map((column) => (
							<th
								key={column}
								className={
									"border-2 border-secondary px-2 py-1 font-bold text-secondary text-xs uppercase"
								}
							>
								{column}
							</th>
						))}
					</tr>
				</thead>
				<tbody>
					<tr>
						{columns.slice(0, 3).map((column, index) => (
							<td
								key={`${index}${column}`}
								className="h-[27px] border border-secondary px-2  text-center font-bold text-2xl text-handwritten leading-none"
							>
								{column}
							</td>
						))}
					</tr>
				</tbody>
			</table>
			<table className="-mt-[1px] w-full table-fixed border-separate border-spacing-0.5">
				<thead>
					<tr>
						{headings.slice(3).map((column) => (
							<th
								key={column}
								className={
									"border-2 border-secondary px-2 py-1 font-bold text-secondary text-xs uppercase"
								}
							>
								{column}
							</th>
						))}
					</tr>
				</thead>
				<tbody>
					<tr>
						{columns.slice(3).map((column, index) => (
							<td
								key={`${index}${column}`}
								className="h-[27px] border border-secondary px-2  text-center font-bold text-2xl text-handwritten leading-none"
							>
								{column}
							</td>
						))}
					</tr>
				</tbody>
			</table>
		</div>
	);
};

export default Header;
