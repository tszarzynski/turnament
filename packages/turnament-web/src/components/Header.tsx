import { AppBar, Theme, Toolbar, Typography } from "@material-ui/core";
import { createStyles, makeStyles } from "@material-ui/styles";
import React from "react";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1
    },
    menuButton: {},
    title: {
      flexGrow: 1
    }
  })
);

export default function Header() {
  const classes = useStyles();
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography variant="h6" className={classes.title}>
          Backgammon Tournament
        </Typography>
      </Toolbar>
    </AppBar>
  );
}
