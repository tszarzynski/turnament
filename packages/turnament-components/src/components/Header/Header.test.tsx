import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';

describe('Header', () => {
  it('renders without error', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
