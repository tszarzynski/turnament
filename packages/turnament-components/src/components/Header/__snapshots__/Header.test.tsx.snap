// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without error 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      >
        Backgammon Tournament
      </h6>
    </div>
  </header>
</DocumentFragment>
`;
